index.html

```diff
@@ -28,89 +28,199 @@
+<br/> &lt;img src=&quot;img/carousel-4.jpg&quot; importance=&quot;low&quot;&gt;
+<br/>&lt;/ul&gt;
+</pre>
+    </div>
+
+    <p>When we assign the off-screen images low priority, this will create less contention between the remaining high priority
+      images and other high priority resources.</p>
+
+    <p>Async JavaScript files may be loaded by default with low priority, but some of them may be of high priority and should
+      be loaded at the same time as the page's render blocking resources. Developers currently use
+      <a href="https://twitter.com/cramforce/status/900445266750263296">hacks</a> to address that use-case.</p>
+    <div class="example" title="Example 1">
+      <p>FastCorp Inc. have a web app with a mix of async scripts that are critical and non-critical to the initial experience.
+        They want to signify the importance of these resources to the browser. They can accomplish this by annotating these
+        requests with the `importance` attribute:
       </p>
       <p>The browser will attempt to fetch main.js with a high priority and the other scripts with a lower priority.</p>
-  <pre highlight="html">
+      <pre highlight="html">
 &lt;script src=&quot;main.js&quot; async importance=&quot;high&quot;&gt;&lt;/script&gt;
 
 &lt;script src=&quot;non-critical-1.js&quot; async importance=&quot;low&quot;&gt;&lt;/script&gt;
 
 &lt;script src=&quot;non-critical-2.js&quot; async importance=&quot;low&quot;&gt;&lt;/script&gt;
   </pre>
-</div>
+    </div>
 
-
-<p>Third party resources (e.g scripts from ads) are often loaded with medium/high priority, but developers may wish to load
-    them all at low priority. Similarly, developers may wish to load all first-party resources that are critical with a high
-    priority.</p>
-<div class="example" title="Example 2">
+    <p>Third party resources (e.g scripts from ads) are often loaded with medium/high priority, but developers may wish to load
+      them all at low priority. Similarly, developers may wish to load all first-party resources that are critical with a
+      high priority.
+    </p>
+    <div class="example" title="Example 2">
       <p>FastCorp Inc. have a page that includes a number of
-         <b>third party</b> resources which are not-critical to first paint. They would like to signify to the browser that the importance
-         of these resources is low so that requests for them don't contend with network requests for more important resources. They can accomplish
-         this by annotating these requests with an `importance` of `low`:
+        <b>third party</b> resources which are not-critical to first paint. They would like to signify to the browser that the
+        importance of these resources is low so that requests for them don't contend with network requests for more important
+        resources. They can accomplish this by annotating these requests with an `importance` of `low`:
```

Comment:
I think this was one of our earlier examples that carried over to this point. I'm in favor of focusing on the importance=high use-case you suggested. Edits will be in the next set of commits.

Reply:
I think this is an important use-case which is not necessarily handled in the current resource loading model for Chrome.
Since scripts are complex, let's consider a third party iframe.
Iframes AFAIR are currently fetched as HTML (so at the highest priority), and there's no way to tell the browser they are not that important.
But... even if we had a way (using priority hints), that currently won't be sufficient, as they would go out to the server on a separate connection, on which they will be the only request and therefore the highest priority one. The same goes for fonts and images. (as well as async scripts - regardless of their H2 priority)

I think Priority Hints is an important building block in solving that important issue. One could also argue that beyond Priority Hints, the rest of the solution is just implementation details

Reply:
(Ah yes, it just occurred to me that it might not be totally clear that my comment was purely talking about scripts being damaging to UX, of course other third-party resource use-cases are very important).

So I want to make sure I have all of the separate-connection stuff right. Assume we have a connection *A* with some requests, and connection *B* with some requests. What I believe now is no matter what we do with the priorities of *B*'s requests (for example with priority hints, making them _all_ low priority), they have no affect on *A*'s request (i.e., because they're separate connections, there will be no inter-connection contention). What I'm a tad fuzzy on is what we can actually do to *B*'s requests to "help out" *A*'s requests, and from what you said on the call today it sounded like the only thing we could do is cut down the number of requests *B* is sending out (some number < what would normally be sent out on that connection), thus giving *A* more bandwidth; is that accurate? If so, it is the queueing strategy that would have the best effect on third-party resources contending with first-party ones it seems.